elide-async/src/main/java/com/yahoo/elide/async/service/thread/AsyncAPIBackgroundRunnable.java

```diff
@@ -0,0 +1,250 @@
+            job.setAsyncApi(exportObj);
+
+            //Notify listeners
+            job.getDone().countDown();
+        }
+    }
+
+    public String generateDownloadURL(TableExport exportObj, RequestScope scope) {
+        String downloadPath =  scope.getElideSettings().getExportApiPath();
+        String baseURL = scope.getBaseUrlEndPoint();
+        return baseURL + downloadPath + "/" + exportObj.getId();
+    }
+
+    protected TableExport storeResults(TableExport exportObj, ResultStorageEngine resultStorageEngine,
+            Observable<String> result) {
+        return resultStorageEngine.storeResults(exportObj, result);
+    }
+
+    private Observable<String> concatStringWithObservable(String toConcat, Observable<String> observable,
+            boolean stringFirst) {
+        if (toConcat == null) {
+            return observable;
+        }
+
+        return stringFirst ? Observable.just(toConcat).concatWith(observable)
+                : observable.concatWith(Observable.just(toConcat));
+    }
+
+    private void validateProjections(Collection<EntityProjection> projections) {
+        List<Validator> validators = new ArrayList<>(Arrays.asList(new SingleRootProjectionValidator()));
+        validators.forEach(validator -> validator.validateProjection(projections));
+    }
+
+    public RequestScope getRequestScope(TableExport export, RequestScope scope, DataStoreTransaction tx,
+            Map<String, List<String>> additionalRequestHeaders) {
+        UUID requestId = UUID.fromString(export.getRequestId());
+        User user = scope.getUser();
+        String apiVersion = scope.getApiVersion();
+        return new GraphQLRequestScope("", tx, user, apiVersion, elide.getElideSettings(),
+                null, requestId, additionalRequestHeaders);
+    }
+
+    public Collection<EntityProjection> getProjections(TableExport export, RequestScope scope) {
+        GraphQLProjectionInfo projectionInfo;
+        try {
+            String graphQLDocument = export.getQuery();
+            ObjectMapper mapper = elide.getMapper().getObjectMapper();
+
+            JsonNode node = QueryRunner.getTopLevelNode(mapper, graphQLDocument);
+            Map<String, Object> variables = QueryRunner.extractVariables(mapper, node);
+            String queryString = QueryRunner.extractQuery(node);
+
+            projectionInfo = new GraphQLEntityProjectionMaker(elide.getElideSettings(), variables,
+                            scope.getApiVersion()).make(queryString);
+
+        } catch (IOException e) {
+            throw new IllegalStateException(e);
+        }
+
+        return projectionInfo.getProjections().values();
+    }
+
+    private Observable<PersistentResource> export(TableExport exportObj, RequestScope scope,
+            EntityProjection projection) {
+        Observable<PersistentResource> results = Observable.empty();
+
+        UUID requestId = UUID.fromString(exportObj.getRequestId());
+
+        try {
```

Comment:
This can be cleaned up since we are restricting async to reads.  We don't need to call precommit, or run any lifecycle hooks for example.